Source/Controllers/SubviewControllers/SPQueryController.m

```diff
@@ -74,7 +74,14 @@ + (SPQueryController *)sharedQueryController
 {
 	@synchronized(self) {
 		if (sharedQueryController == nil) {
-			sharedQueryController = [[super allocWithZone:NULL] init];
+			if (![NSThread isMainThread]){
+				SPMainQSync(^{
+					sharedQueryController = [[super allocWithZone:NULL] init];
```

Comment:
not sure about that... why can't we just `[[SPQueryController alloc] init]`?

Reply:
goes into some weird loop..

Reply:
we can remove allocWithZone:(NSZone *)zone and then there's no loop.